### PR TITLE
Fix cross-org isolation for agents, signed URLs, and local storage.

### PR DESCRIPTION
**Pull request by Pranjal-wq**
Service-account chat and the SA probe now require the caller's org. Signed-URL mint and download check JWT identity, record org, and ACL. Local storage rejects any path that leaves the mount.

## Description

ssue 1 — Org B could chat as Org A’s service agent
File: backend/python/app/api/routes/agent.py

Bug: For a service-account agent, chat skipped the usual permission check and ran as the person who created the agent. Org B only needed the agent id.

Fix: After loading the creator, we now check: caller’s company == creator’s company. If not → “agent not found.” Same-company Slack still works.

Issue 2 — Org A could overwrite Org B’s file on disk
File: backend/nodejs/apps/src/modules/storage/providers/local-storage.provider.ts
(function: sanitizePath)

Bug: The path cleaner only removed ../ at the start. A path like orgA/PipesHub/../../orgB/PipesHub/secret.pdf still walked into Org B’s folder.

Fix: Any .. in the path is rejected. The file must stay inside that company’s folder.

Issue 3 — Org B could learn if Org A’s agent exists / is a service account
File: same as issue 1 — backend/python/app/api/routes/agent.py
(function: _assert_agent_belongs_to_caller_org, used by get_agent_internal)

Bug: A special “is this a service account?” API answered differently: missing = 404, exists but not SA = 403, is SA = 200. Org B could guess ids.

Fix: If the agent is not in the caller’s company, always 404. Same answer whether it exists or not.

Issue 4 — “Shared with me” chats did not filter by company
Files:

backend/nodejs/apps/src/modules/enterprise_search/utils/utils.ts (buildAgentSharedWithMeFilter)
backend/nodejs/apps/src/modules/enterprise_search/controller/es_controller.ts (passes orgId in)
Bug: The query only checked “shared with this user,” not “same company.” Nothing writes that share list today, so it was a future hole.

Fix: The filter now also requires the caller’s orgId.

Issue 5 — Org B could download Org A’s file with only a record id
File: backend/python/app/connectors/api/router.py

Bug: Two steps:

Mint a download link — trusted the company/user in the URL, not the logged-in user, and skipped permission checks.
Download — only checked the token matched the record id, not the company.
Fix:

Mint: URL company/user must match the login; the file must belong to that company; the user must have access. Otherwise 404.
Download: the file’s company must match the URL (and the token’s company, if present).
Tests (same ideas, just proving it):
test_agent.py, test_agent_full_coverage.py, local-storage.provider.test.ts, utils.test.ts, test_router_part1.py

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Bug Fixes

- Shared agent conversations are now scoped to the caller’s organization.
- Agent chat and internal lookups reject cross-organization access.
- Signed URL generation and file downloads validate organization ownership and permissions.
- Scoped service tokens can securely access authorized files without requiring a user ID.
- Local storage rejects unsafe, empty, absolute, and directory-traversal paths.
- Invalid cross-organization or unauthorized requests now return appropriate errors.

### CODE REVIEW

File: backend/nodejs/apps/src/modules/enterprise_search/utils/utils.ts [hunk 1]
```diff
@@ -1339,11 +1339,13 @@ export const buildAgentConversationFilter = (
  */
 export const buildAgentSharedWithMeFilter = (
   req: any,
+  orgId: string,
   userId: string,
   agentKey: string,
 ) => {
   const filter: any = {
```
[the remark this answers is not in the document]
Comment (Pranjal-wq) [reply backend/nodejs/apps/src/modules/enterprise_search/utils/utils.ts:1346]: t matches buildAgentConversationFilter next to it. Typing only this one is churn; we can type both filters in a follow-up.

File: backend/nodejs/apps/tests/modules/storage/providers/local-storage.provider.test.ts [hunk 2]
```diff
@@ -147,10 +147,9 @@ describe('LocalStorageAdapter', () => {
   // sanitizePath (private)
   // -------------------------------------------------------------------------
   describe('sanitizePath (private)', () => {
-    it('should remove parent directory references', () => {
+    it('should reject leading parent-directory segments', () => {
       const adapter = createAdapter()
-      const result = (adapter as any).sanitizePath('../../etc/passwd')
-      expect(result).to.not.include('..')
+      expect(() => (adapter as any).sanitizePath('../../etc/passwd')).to.throw(StorageValidationError)
```
[the remark this answers is not in the document]
Comment (Pranjal-wq) [reply backend/nodejs/apps/tests/modules/storage/providers/local-storage.provider.test.ts:152]: ame cast as the existing sanitizePath tests in this file (private method). Not introducing a new pattern. No production impact.

File: backend/nodejs/apps/src/modules/storage/providers/local-storage.provider.ts [hunk 3]
```diff
@@ -414,12 +424,48 @@
...
-    // Remove any parent directory references for security
-    const normalizedPath = path
-      .normalize(filePath)
-      .replace(/^(\.\.[\/\\])+/, '');
-    return normalizedPath;
+    if (!filePath || filePath.includes('\0')) {
+      throw new StorageValidationError('Invalid document path');
+    }
+
+    const unified = filePath.replace(/\\/g, '/').replace(/\/+$/, '');
```
[the remark this answers is not in the document]
Comment (Pranjal-wq) [reply backend/nodejs/apps/src/modules/storage/providers/local-storage.provider.ts]: done

**Review by tushar1245 — APPROVED** [review]

**Review by abhishek-pipeshub — APPROVED** [review]
